tests/imputers/test_ModeImputer.py

```diff
@@ -159,201 +134,22 @@ def test_fit_not_changing_data_weighted(self):
-        x.impute_values_ = {"a": 1.0, "b": 1.0, "c": -6.0}
-
-        df_transformed = x.transform(df)
-
-        ta.equality.assert_equal_dispatch(
-            expected=expected,
-            actual=df_transformed,
-            msg="Check nulls filled correctly in transform",
-        )
-
-    @pytest.mark.parametrize(
-        ("df", "expected"),
-        ta.pandas.adjusted_dataframe_params(d.create_df_3(), expected_df_2()),
-    )
-    def test_nulls_imputed_correctly_2(self, df, expected):
-        """Test missing values are filled with the correct values - and unrelated columns are not changed."""
-        x = ModeImputer(columns=["a"])
-
-        # set the impute values dict directly rather than fitting x on df so test works with helpers
-        x.impute_values_ = {"a": 1.0}
-
-        df_transformed = x.transform(df)
-
-        ta.equality.assert_equal_dispatch(
-            expected=expected,
-            actual=df_transformed,
-            msg="Check nulls filled correctly in transform",
-        )
-
-    @pytest.mark.parametrize(
-        ("df", "expected"),
-        ta.pandas.row_by_row_params(d.create_df_9(), expected_df_3())
-        + ta.pandas.index_preserved_params(d.create_df_9(), expected_df_3()),
-    )
-    def test_nulls_imputed_correctly_3(self, df, expected):
-        """Test missing values are filled with the correct values - and unrelated columns are not changed
-        (when weight is used).
-        """
-        x = ModeImputer(columns=["a"], weight="c")
-
-        # set the impute values dict directly rather than fitting x on df so test works with helpers
-        x.impute_values_ = {"a": 6}
-
-        df_transformed = x.transform(df)
-
-        ta.equality.assert_equal_dispatch(
-            expected=expected,
-            actual=df_transformed,
-            msg="Check nulls filled correctly in transform",
-        )
-
-    def test_learnt_values_not_modified(self):
-        """Test that the impute_values_ from fit are not changed in transform."""
-        df = d.create_df_3()
 
-        x = ModeImputer(columns=["a", "b", "c"])
+class TestTransform(GenericTransformTests, GenericImputerTransformTests):
+    """Tests for transformer.transform."""
 
-        x.fit(df)
+    @classmethod
+    def setup_class(cls):
+        cls.transformer_name = "ModeImputer"
 
-        x2 = ModeImputer(columns=["a", "b", "c"])
 
-        x2.fit_transform(df)
+class TestOtherBaseBehaviour(OtherBaseBehaviourTests):
+    """
+    Class to run tests for BaseTransformerBehaviour outside the three standard methods.
 
-        ta.equality.assert_equal_dispatch(
-            expected=x.impute_values_,
-            actual=x2.impute_values_,
-            msg="Impute values not changed in transform",
-        )
+    May need to overwite specific tests in this class if the tested transformer modifies this behaviour.
+    """
 
-    def test_learnt_values_not_modified_weights(self):
```

Comment:
worth reintroducing this one for the time being too

Reply:
No probs